File: src/main/java/ar/edu/um/tesoreria/rest/model/CargoMateria.java
/**
 * 
 */
package ar.edu.um.tesoreria.rest.model;

import java.io.Serializable;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Table(name = "cargomateria")
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
public class CargoMateria extends Auditable implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = 4197006793534538460L;

	@Id
	@Column(name = "cam_id")
	private Integer cargomateriaId;

	@Column(name = "cam_nombre")
	private String nombre = "";

	@Column(name = "cam_precedencia")
	private Integer precedencia = 0;

}

File: src/main/java/um/tesoreria/rest/kotlin/model/CargoMateria.kt
package um.tesoreria.rest.kotlin.model

import um.tesoreria.rest.model.Auditable
import jakarta.persistence.Entity
import jakarta.persistence.Table
import jakarta.persistence.Id
import jakarta.persistence.Column


@Entity
@Table(name = "cargomateria")
data class CargoMateria(

	@Id
	@Column(name = "cam_id")
	var cargomateriaId: Int? = null,

	@Column(name = "cam_nombre")
	var nombre: String = "",

	@Column(name = "cam_precedencia")
	var precedencia: Int = 0

) : Auditable()